jackson-jq/src/main/java/net/thisptr/jackson/jq/ObjectMapperDefaultProvider.java

```diff
@@ -0,0 +1,20 @@
+package net.thisptr.jackson.jq;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import net.thisptr.jackson.jq.internal.misc.JsonQueryJacksonModule;
+
+/**
+ * Default {@link ObjectMapperProvider}
+ */
+public class ObjectMapperDefaultProvider implements ObjectMapperProvider {
+
```

Comment:
is ObjectMapperHolder class really needed, can we just have 
private static final ObjectMapper INSTANCE = new ObjectMapper().registerModule(JsonQueryJacksonModule.getInstance()); and get() method doing "return INSTANCE"?

Reply:
Hi! This is a [lazy-loaded singleton](https://en.wikipedia.org/wiki/Initialization-on-demand_holder_idiom). So we won't have unnecessary `ObjectMapper`s in memory, especially in a CDI context, whereas the platform can provide a reference for the `ObjectMapper`.